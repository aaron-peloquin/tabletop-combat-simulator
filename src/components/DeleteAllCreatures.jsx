import {connect} from "react-redux"
import {Button} from "@material-ui/core"

import deleteAll from "./../store/action/DeleteAllCreatures"

const DeleteAllCreatures = (props) => {
  const {ActionDeleteAll} = props
  return <Button data-id="delete-all" onClick={ActionDeleteAll} variant="contained" color="secondary">
    Delete All Creatures
  </Button>
}

const MapStateToProps = () => {
  return { }
}

const MapActionsToProps = (dispatch) => {
  return {
    ActionDeleteAll: () => {
      deleteAll(dispatch, "delete all")
    },
  }
}

export default connect(MapStateToProps, MapActionsToProps)(DeleteAllCreatures)
