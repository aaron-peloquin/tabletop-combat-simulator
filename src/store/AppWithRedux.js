import React from 'react'
import {initializeStore} from './store'

const isServer = typeof window === 'undefined'
const reduxStoreSymbol = Symbol("reduxStore")

function getOrCreateStore (initialState) {
  // Always make a new store if server, otherwise state is shared between requests
  if (isServer) {
    return initializeStore(initialState)
  }

  // Create store if unavailable on the client and set it on the window object
  if (!window[reduxStoreSymbol]) {
    window[reduxStoreSymbol] = initializeStore(initialState)
  }
  return window[reduxStoreSymbol]
}

export default (App) => {
  return class AppWithRedux extends React.Component {
    static async getInitialProps (appContext) {
      // Get or Create the store with `undefined` as initialState
      // This allows you to set a custom default initialState
      const reduxStore = getOrCreateStore()

      // Provide the store to getInitialProps of pages
      appContext.ctx.reduxStore = reduxStore

      let appProps = {}
      if (typeof App.getInitialProps === 'function') {
        appProps = await App.getInitialProps(appContext)
      }

      return {
        ...appProps,
        initialReduxState: reduxStore.getState()
      }
    }

    constructor (props) {
      super(props)
      this.reduxStore = getOrCreateStore(props.initialReduxState)
    }

    render () {
      return <App {...this.props} reduxStore={this.reduxStore} />
    }
  }
}
