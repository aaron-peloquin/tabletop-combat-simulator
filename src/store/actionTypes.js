/**
 * This is an object of Redux Actions. It mostly exists to
 *  ensure we do not create duplicate actions, but it can
 *  also be used as a list of our existing actions
*/
const actions = {
  CopyCreature: 'CopyCreature',
  DeleteAllCreatures: 'DeleteAllCreatures',
  DeleteCreature: 'DeleteCreature',
  SaveCreature: 'SaveCreature',
  SetEditCreature: 'SetEditCreature',
  ToggleSidebar: 'ToggleSidebar',
  RunSimulation: 'RunSimulation',
  UpdateEditCreature: 'UpdateEditCreature',
}

export default actions
