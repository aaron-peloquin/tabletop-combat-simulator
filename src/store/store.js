import {createStore, applyMiddleware, combineReducers} from "redux"
import {composeWithDevTools} from "redux-devtools-extension"
import thunkMiddleware from "redux-thunk"

/** Reducers */
import combat from "./reducer/combat"
import creatures from "./reducer/creatures"
import sideBar from "./reducer/sideBar"
import editing from "./reducer/editing"

/** Reducers */
const reducers = combineReducers({
  combat,
  creatures,
  editing,
  sideBar,
})

/**
 * Initializes the store
 * @param {obj} initialState the state our app starts with
 * @return {obj} the redux store
 */
export function initializeStore(initialState = initialState) {
  return createStore(reducers, initialState, composeWithDevTools(applyMiddleware(thunkMiddleware)))
}
