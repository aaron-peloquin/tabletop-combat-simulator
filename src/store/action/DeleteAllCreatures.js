import actionTypes from '../actionTypes'

/**
 * WARNING: Deletes all creature data
 * @param {func} dispatch the redux dispatcher action
 * @param {str} confirm (safety) must be set to "delete all", or this will not run
 * @return {void}
 */
const DeleteAllCreatures = (dispatch, confirm) => {
  if (confirm==='delete all') {
    const data = {
      type: actionTypes.DeleteAllCreatures,
      payload: {},
    }
    dispatch(data)
  }
}

export default DeleteAllCreatures
