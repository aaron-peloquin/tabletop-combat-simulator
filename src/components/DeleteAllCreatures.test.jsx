import { mount } from "enzyme"
import { Provider } from "react-redux"
import toJson from "enzyme-to-json"

import {initializeStore} from "../store/store"
import DeleteAllCreatures from "../components/DeleteAllCreatures"

describe("<DeleteAllCreatures />", ()=>{
  let store, props, DeleteAllCreaturesComponent
  beforeEach(async ()=>{
    store = initializeStore({})
    store.dispatch = jest.fn(store.dispatch)
    props = await DeleteAllCreatures.getInitialProps()
    props.ActionDeleteAll = jest.fn(props.ActionDeleteAll)
    DeleteAllCreaturesComponent = mount(<Provider store={store}><DeleteAllCreatures {...props} /></Provider>).find("DeleteAllCreatures")
  })

  it("calls dispatch on click", () => {
    DeleteAllCreaturesComponent.simulate("click")
    expect(props.ActionDeleteAll).toBeCalledTimes(1)
    expect(store.dispatch).toBeCalledTimes(1)
  })

  it("snapshots", () => {
    const Tree = toJson(DeleteAllCreaturesComponent)
    expect(Tree).toMatchSnapshot()
  })
})