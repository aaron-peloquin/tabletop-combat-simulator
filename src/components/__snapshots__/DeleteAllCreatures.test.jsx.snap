// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteAllCreatures /> snapshots 1`] = `
<WithStyles(Button)
  color="secondary"
  data-id="delete-all"
  onClick={[Function]}
  variant="contained"
>
  Delete All Creatures
</WithStyles(Button)>
`;
